### ISSUE TITLE
Save State Menu Request

### ISSUE THREAD

**Issue by miroswan** [mamedev/mame#2642]
It would be really helpful to be able to save a save state in the UI without using hot keys. If you're using MAME with a joystick, you don't want to have to reach for a keyboard to save and load states. Also, it would be great if frontends had the capability of controlling save states.

**Comment by cuavas** [comment]
You can save and load states with a joystick.  You need to map buttons or button combinations to the "Save State" and "Load State" inputs, but once you're on the menu you can press a joystick button to save or load machine state.

**Comment by miroswan** [comment]
Yes, but then you're limited by the number of buttons on your joystick. Is it all at feasible to get a configurable number slots to which you can save a state within the UI?

**Comment by miroswan** [comment]
If it's feasible and someone can point me in the right direction maybe I can help. Could be a fun little side project.
